lake/attributes/formal_attr.py

```diff
@@ -38,3 +39,39 @@ def get_port_name(self):
 
     def get_formal_ann(self):
         return self.formalsig_cnst
+
+
+# these attributes should be used for all modules (agg, sram, tb)
+class FormalAttr(FormalAttrBase):
```

Comment:
Do these need to be separate classes? Seems they could just have a different string in the constructor.

Reply:
nope, they don't! I was thinking about doing some stuff in there initially but abandoned that so just changed it a string parameter like you said.